src/nodes/Components/MetaMask.js

```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
 }
 
 MetaMask.prototype.onAction = async function() {
-  try{
-    console.log("CALLING ENABLE ON METAMASK!!!")
-    this.accounts = await window.ethereum.enable()
-  }catch(e){
-    console.log(e)
-  }
+  this.connectWeb3()
 }
 
-MetaMask.prototype.connectWeb3 = function() {
-   if (typeof window.web3 !== 'undefined'){
-     this.web3 = window.web3
-      console.log('MetaMask is installed',this.web3)
-   }
-   else{
-      console.log('MetaMask is not installed')
-   }
+MetaMask.prototype.connectWeb3 = async function() {
+  const provider = detectEthereumProvider
+  if (provider) {
+    try {
+      this.accounts = await window.ethereum.request({ method: 'eth_requestAccounts' })
+      this.setOutputData(0, this.accounts[0])
+    } catch (error) {
+        if (error.code === 4001) {
+          console.log('Please connect to MetaMask!')
+        } else {
+          console.error(error)
+        }
+    }
+  } else {
+    console.log('Please install MetaMask!')
+  }
 }
 
 MetaMask.prototype.onExecute = async function() {
-
-  if(window.web3 && window.web3.eth && typeof window.web3.eth.getAccounts == "function"){
-    window.web3.eth.getAccounts((error,accounts)=>{
-      this.accounts = accounts
-    })
-  }
-  if(this.accounts){
-    this.setOutputData(0,this.accounts[0])
-  }
-
+  this.connectWeb3()
 
   this.setOutputData(1,{
     name:"balance",
-    args:[{name:"address",type:"string"}],
-    function:async (args)=>{
+    function:async ()=>{
```

Comment:
Changed this to no longer get inputs from user and only use the current selected MM address.